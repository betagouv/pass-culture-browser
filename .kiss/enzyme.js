import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
