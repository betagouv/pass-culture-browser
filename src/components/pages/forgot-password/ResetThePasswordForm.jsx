import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import { FormError, FormFooter } from '../../forms'
import HiddenField from '../../forms/inputs/HiddenField'
import PasswordField from '../../forms/inputs/PasswordField'
import withResetForm from './withResetForm'

const cancelOptions = {
  className: 'is-white-text',
  disabled: false,
  label: 'Annuler',
  url: '/connexion',
}

const submitOptions = {
  className: 'is-bold is-white-text',
  label: 'OK',
}

export const ResetThePasswordForm = ({ canSubmit, formErrors, isLoading }) => (
  <Fragment>
    <div>
      <div className="fs22">
        <h2 className="is-italic is-medium">
          <span className="is-block">{'Saisissez ci-dessous'}</span>
          <span className="is-block">{'votre nouveau mot de passe.'}</span>
        </h2>
        <p className="mt12 fs16">
          {
            'Il doit contenir au minimum 12 caractères, une majuscule, une minuscule, un chiffre et un caractère spécial.'
          }
        </p>
        <p className="is-block is-regular fs13 mt18">
          <span>{'*'}</span>
          &nbsp;{'Champs obligatoires'}
        </p>
      </div>
      <div>
        <PasswordField
          disabled={isLoading}
          label="Saisissez votre nouveau mot de passe"
          name="newPassword"
          required
          theme="primary"
        />
        <PasswordField
          disabled={isLoading}
          label="Confirmez votre nouveau mot de passe"
          name="newPasswordConfirm"
          required
          theme="primary"
        />
        <HiddenField name="token" />
        {formErrors && <FormError customMessage={formErrors} />}
      </div>
    </div>
    <FormFooter
      cancel={cancelOptions}
      submit={{ ...submitOptions, disabled: !canSubmit }}
    />
  </Fragment>
)

ResetThePasswordForm.defaultProps = {
  formErrors: false,
}

ResetThePasswordForm.propTypes = {
  canSubmit: PropTypes.bool.isRequired,
  formErrors: PropTypes.oneOfType([PropTypes.array, PropTypes.bool, PropTypes.string]),
  isLoading: PropTypes.bool.isRequired,
}

export default withResetForm(ResetThePasswordForm, null, '/users/new-password', 'POST')
