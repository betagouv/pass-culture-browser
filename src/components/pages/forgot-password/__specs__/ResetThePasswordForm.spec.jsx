import React from 'react'
import { shallow } from 'enzyme'
import { ResetThePasswordForm } from '../ResetThePasswordForm'

describe('src | components | pages | forgot-password | ResetThePasswordForm', () => {
  it('should match the snapshot', () => {
    // given
    const props = { canSubmit: true, isLoading: false }

    // when
    const wrapper = shallow(<ResetThePasswordForm {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })
})
