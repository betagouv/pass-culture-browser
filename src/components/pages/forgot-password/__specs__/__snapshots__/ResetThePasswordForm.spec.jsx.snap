// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | forgot-password | ResetThePasswordForm should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    canSubmit={true}
    formErrors={false}
    isLoading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div>
          <div
            className="fs22"
          >
            <h2
              className="is-italic is-medium"
            >
              <span
                className="is-block"
              >
                Saisissez ci-dessous
              </span>
              <span
                className="is-block"
              >
                votre nouveau mot de passe.
              </span>
            </h2>
            <p
              className="mt12 fs16"
            >
              Il doit contenir au minimum 12 caractères, une majuscule, une minuscule, un chiffre et un caractère spécial.
            </p>
            <p
              className="is-block is-regular fs13 mt18"
            >
              <span>
                *
              </span>
               
              Champs obligatoires
            </p>
          </div>
          <div>
            <PasswordField
              autoComplete={false}
              disabled={false}
              id={null}
              label="Saisissez votre nouveau mot de passe"
              name="newPassword"
              placeholder=""
              required={true}
              sublabel={null}
              theme="primary"
            />
            <PasswordField
              autoComplete={false}
              disabled={false}
              id={null}
              label="Confirmez votre nouveau mot de passe"
              name="newPasswordConfirm"
              placeholder=""
              required={true}
              sublabel={null}
              theme="primary"
            />
            <HiddenField
              name="token"
              validator={[Function]}
            />
          </div>
        </div>,
        <FormFooter
          cancel={
            Object {
              "className": "is-white-text",
              "disabled": false,
              "label": "Annuler",
              "url": "/connexion",
            }
          }
          className=""
          externalLink={null}
          submit={
            Object {
              "className": "is-bold is-white-text",
              "disabled": false,
              "label": "OK",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="fs22"
            >
              <h2
                className="is-italic is-medium"
              >
                <span
                  className="is-block"
                >
                  Saisissez ci-dessous
                </span>
                <span
                  className="is-block"
                >
                  votre nouveau mot de passe.
                </span>
              </h2>
              <p
                className="mt12 fs16"
              >
                Il doit contenir au minimum 12 caractères, une majuscule, une minuscule, un chiffre et un caractère spécial.
              </p>
              <p
                className="is-block is-regular fs13 mt18"
              >
                <span>
                  *
                </span>
                 
                Champs obligatoires
              </p>
            </div>,
            <div>
              <PasswordField
                autoComplete={false}
                disabled={false}
                id={null}
                label="Saisissez votre nouveau mot de passe"
                name="newPassword"
                placeholder=""
                required={true}
                sublabel={null}
                theme="primary"
              />
              <PasswordField
                autoComplete={false}
                disabled={false}
                id={null}
                label="Confirmez votre nouveau mot de passe"
                name="newPasswordConfirm"
                placeholder=""
                required={true}
                sublabel={null}
                theme="primary"
              />
              <HiddenField
                name="token"
                validator={[Function]}
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="is-italic is-medium"
                >
                  <span
                    className="is-block"
                  >
                    Saisissez ci-dessous
                  </span>
                  <span
                    className="is-block"
                  >
                    votre nouveau mot de passe.
                  </span>
                </h2>,
                <p
                  className="mt12 fs16"
                >
                  Il doit contenir au minimum 12 caractères, une majuscule, une minuscule, un chiffre et un caractère spécial.
                </p>,
                <p
                  className="is-block is-regular fs13 mt18"
                >
                  <span>
                    *
                  </span>
                   
                  Champs obligatoires
                </p>,
              ],
              "className": "fs22",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="is-block"
                    >
                      Saisissez ci-dessous
                    </span>,
                    <span
                      className="is-block"
                    >
                      votre nouveau mot de passe.
                    </span>,
                  ],
                  "className": "is-italic is-medium",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Saisissez ci-dessous",
                      "className": "is-block",
                    },
                    "ref": null,
                    "rendered": "Saisissez ci-dessous",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "votre nouveau mot de passe.",
                      "className": "is-block",
                    },
                    "ref": null,
                    "rendered": "votre nouveau mot de passe.",
                    "type": "span",
                  },
                ],
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Il doit contenir au minimum 12 caractères, une majuscule, une minuscule, un chiffre et un caractère spécial.",
                  "className": "mt12 fs16",
                },
                "ref": null,
                "rendered": "Il doit contenir au minimum 12 caractères, une majuscule, une minuscule, un chiffre et un caractère spécial.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      *
                    </span>,
                    " ",
                    "Champs obligatoires",
                  ],
                  "className": "is-block is-regular fs13 mt18",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "*",
                    },
                    "ref": null,
                    "rendered": "*",
                    "type": "span",
                  },
                  " ",
                  "Champs obligatoires",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <PasswordField
                  autoComplete={false}
                  disabled={false}
                  id={null}
                  label="Saisissez votre nouveau mot de passe"
                  name="newPassword"
                  placeholder=""
                  required={true}
                  sublabel={null}
                  theme="primary"
                />,
                <PasswordField
                  autoComplete={false}
                  disabled={false}
                  id={null}
                  label="Confirmez votre nouveau mot de passe"
                  name="newPasswordConfirm"
                  placeholder=""
                  required={true}
                  sublabel={null}
                  theme="primary"
                />,
                <HiddenField
                  name="token"
                  validator={[Function]}
                />,
                false,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "autoComplete": false,
                  "disabled": false,
                  "id": null,
                  "label": "Saisissez votre nouveau mot de passe",
                  "name": "newPassword",
                  "placeholder": "",
                  "required": true,
                  "sublabel": null,
                  "theme": "primary",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "autoComplete": false,
                  "disabled": false,
                  "id": null,
                  "label": "Confirmez votre nouveau mot de passe",
                  "name": "newPasswordConfirm",
                  "placeholder": "",
                  "required": true,
                  "sublabel": null,
                  "theme": "primary",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "token",
                  "validator": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              false,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cancel": Object {
            "className": "is-white-text",
            "disabled": false,
            "label": "Annuler",
            "url": "/connexion",
          },
          "className": "",
          "externalLink": null,
          "submit": Object {
            "className": "is-bold is-white-text",
            "disabled": false,
            "label": "OK",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div>
            <div
              className="fs22"
            >
              <h2
                className="is-italic is-medium"
              >
                <span
                  className="is-block"
                >
                  Saisissez ci-dessous
                </span>
                <span
                  className="is-block"
                >
                  votre nouveau mot de passe.
                </span>
              </h2>
              <p
                className="mt12 fs16"
              >
                Il doit contenir au minimum 12 caractères, une majuscule, une minuscule, un chiffre et un caractère spécial.
              </p>
              <p
                className="is-block is-regular fs13 mt18"
              >
                <span>
                  *
                </span>
                 
                Champs obligatoires
              </p>
            </div>
            <div>
              <PasswordField
                autoComplete={false}
                disabled={false}
                id={null}
                label="Saisissez votre nouveau mot de passe"
                name="newPassword"
                placeholder=""
                required={true}
                sublabel={null}
                theme="primary"
              />
              <PasswordField
                autoComplete={false}
                disabled={false}
                id={null}
                label="Confirmez votre nouveau mot de passe"
                name="newPasswordConfirm"
                placeholder=""
                required={true}
                sublabel={null}
                theme="primary"
              />
              <HiddenField
                name="token"
                validator={[Function]}
              />
            </div>
          </div>,
          <FormFooter
            cancel={
              Object {
                "className": "is-white-text",
                "disabled": false,
                "label": "Annuler",
                "url": "/connexion",
              }
            }
            className=""
            externalLink={null}
            submit={
              Object {
                "className": "is-bold is-white-text",
                "disabled": false,
                "label": "OK",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="fs22"
              >
                <h2
                  className="is-italic is-medium"
                >
                  <span
                    className="is-block"
                  >
                    Saisissez ci-dessous
                  </span>
                  <span
                    className="is-block"
                  >
                    votre nouveau mot de passe.
                  </span>
                </h2>
                <p
                  className="mt12 fs16"
                >
                  Il doit contenir au minimum 12 caractères, une majuscule, une minuscule, un chiffre et un caractère spécial.
                </p>
                <p
                  className="is-block is-regular fs13 mt18"
                >
                  <span>
                    *
                  </span>
                   
                  Champs obligatoires
                </p>
              </div>,
              <div>
                <PasswordField
                  autoComplete={false}
                  disabled={false}
                  id={null}
                  label="Saisissez votre nouveau mot de passe"
                  name="newPassword"
                  placeholder=""
                  required={true}
                  sublabel={null}
                  theme="primary"
                />
                <PasswordField
                  autoComplete={false}
                  disabled={false}
                  id={null}
                  label="Confirmez votre nouveau mot de passe"
                  name="newPasswordConfirm"
                  placeholder=""
                  required={true}
                  sublabel={null}
                  theme="primary"
                />
                <HiddenField
                  name="token"
                  validator={[Function]}
                />
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2
                    className="is-italic is-medium"
                  >
                    <span
                      className="is-block"
                    >
                      Saisissez ci-dessous
                    </span>
                    <span
                      className="is-block"
                    >
                      votre nouveau mot de passe.
                    </span>
                  </h2>,
                  <p
                    className="mt12 fs16"
                  >
                    Il doit contenir au minimum 12 caractères, une majuscule, une minuscule, un chiffre et un caractère spécial.
                  </p>,
                  <p
                    className="is-block is-regular fs13 mt18"
                  >
                    <span>
                      *
                    </span>
                     
                    Champs obligatoires
                  </p>,
                ],
                "className": "fs22",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span
                        className="is-block"
                      >
                        Saisissez ci-dessous
                      </span>,
                      <span
                        className="is-block"
                      >
                        votre nouveau mot de passe.
                      </span>,
                    ],
                    "className": "is-italic is-medium",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Saisissez ci-dessous",
                        "className": "is-block",
                      },
                      "ref": null,
                      "rendered": "Saisissez ci-dessous",
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "votre nouveau mot de passe.",
                        "className": "is-block",
                      },
                      "ref": null,
                      "rendered": "votre nouveau mot de passe.",
                      "type": "span",
                    },
                  ],
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Il doit contenir au minimum 12 caractères, une majuscule, une minuscule, un chiffre et un caractère spécial.",
                    "className": "mt12 fs16",
                  },
                  "ref": null,
                  "rendered": "Il doit contenir au minimum 12 caractères, une majuscule, une minuscule, un chiffre et un caractère spécial.",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <span>
                        *
                      </span>,
                      " ",
                      "Champs obligatoires",
                    ],
                    "className": "is-block is-regular fs13 mt18",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "*",
                      },
                      "ref": null,
                      "rendered": "*",
                      "type": "span",
                    },
                    " ",
                    "Champs obligatoires",
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <PasswordField
                    autoComplete={false}
                    disabled={false}
                    id={null}
                    label="Saisissez votre nouveau mot de passe"
                    name="newPassword"
                    placeholder=""
                    required={true}
                    sublabel={null}
                    theme="primary"
                  />,
                  <PasswordField
                    autoComplete={false}
                    disabled={false}
                    id={null}
                    label="Confirmez votre nouveau mot de passe"
                    name="newPasswordConfirm"
                    placeholder=""
                    required={true}
                    sublabel={null}
                    theme="primary"
                  />,
                  <HiddenField
                    name="token"
                    validator={[Function]}
                  />,
                  false,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "autoComplete": false,
                    "disabled": false,
                    "id": null,
                    "label": "Saisissez votre nouveau mot de passe",
                    "name": "newPassword",
                    "placeholder": "",
                    "required": true,
                    "sublabel": null,
                    "theme": "primary",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "autoComplete": false,
                    "disabled": false,
                    "id": null,
                    "label": "Confirmez votre nouveau mot de passe",
                    "name": "newPasswordConfirm",
                    "placeholder": "",
                    "required": true,
                    "sublabel": null,
                    "theme": "primary",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "name": "token",
                    "validator": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                false,
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cancel": Object {
              "className": "is-white-text",
              "disabled": false,
              "label": "Annuler",
              "url": "/connexion",
            },
            "className": "",
            "externalLink": null,
            "submit": Object {
              "className": "is-bold is-white-text",
              "disabled": false,
              "label": "OK",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
